signac/core/synced_collection.py

```diff
@@ -30,6 +31,7 @@ class SyncedCollection(Collection):
     """
 
     backend = None
+    _validators: List[object] = list()  # list of callable objects
```

Comment:
Have we been typing in other places in the recent PRs?

Reply:
No, I got an [error](https://app.circleci.com/pipelines/github/glotzerlab/signac/1300/workflows/64d8599a-d2c3-4f0e-b5ad-63dbb194d158/jobs/8088) while testing on CI for this case.

Reply:
Got it, I don't recall when exactly we added mypy validation but makes sense.